src/main/java/address/controller/MainController.java

```diff
@@ -575,9 +560,29 @@ private void minimizeWindow() {
         primaryStage.setMaximized(false);
     }
 
-    private void maximizeWindow() {
-        primaryStage.setMaximized(true);
-        primaryStage.setIconified(false);
+    private void resizeWindow() {
+        logger.info("Resizing window");
+        if (primaryStage.isIconified()) {
+            primaryStage.setIconified(false);
+        } else {
+            // specially handle since stage operations on Mac seem to not be working as intended
+            if (OsDetector.isOnMac()) {
+                // refresh stage so that resizing effects (apart from the first resize after iconify-ing) are applied
+                // however, this will cause minor flinching in window visibility
+                primaryStage.hide(); // hide has to be called before setMaximized,
+                                     // or first attempt after iconify-ing will resize twice
+                primaryStage.show();
+
+                // on Mac, setMaximized seems to work like "setResize"
+                // isMaximized also does not seem to return the correct value
+                primaryStage.setMaximized(true);
+            } else {
+                primaryStage.setMaximized(!primaryStage.isMaximized());
+            }
+        }
+
+        logger.debug("Stage width: {}", primaryStage.getWidth());
+        logger.debug("Stage height: {}", primaryStage.getHeight());
```

Comment:
I don't like the 'pregnant woman' shape (caused by short branch followed by deep branch) of the code, but not sure if it can be avoided.
```
\
/
\
 \
  \
   \
   /
  /
 /
/
```